[Bug] Missing pydantic dependency causes MCP test import failure

## Summary

The MCP (Model Context Protocol) tests fail to import due to missing `pydantic` dependency.

## Error

```
tests/test_mcp_session_manager.py:12: in <module>
    from kicad_tools.mcp.errors import (
src/kicad_tools/mcp/errors.py:9: in <module>
    from pydantic import BaseModel
E   ModuleNotFoundError: No module named 'pydantic'
```

## Observed Behavior

Running `uv run pytest tests/` fails immediately with import error before any tests can run.

## Expected Behavior

All dependencies should be properly declared and installed for tests to run.

## Reproduction

```bash
uv run pytest tests/test_mcp_session_manager.py
```

## Suggested Fix

1. Add `pydantic` to the project dependencies in `pyproject.toml`
2. Or add it as a dev/test dependency if only needed for testing
3. Alternatively, make MCP tests optional if pydantic is an optional dependency

## Impact

- Full test suite cannot be run
- CI may be silently skipping these tests
- MCP functionality may be broken in production

Discovered during review of PR #696

--- a/tests/test_mcp_analysis.py
+++ b/tests/test_mcp_analysis.py
@@ -6,6 +6,8 @@ from pathlib import Path
 
 import pytest
 
+pytest.importorskip("pydantic")
+
 from kicad_tools.exceptions import FileNotFoundError as KiCadFileNotFoundError
 from kicad_tools.exceptions import ParseError
 from kicad_tools.mcp.tools.analysis import analyze_board

--- a/tests/test_mcp_assembly.py
+++ b/tests/test_mcp_assembly.py
@@ -6,6 +6,10 @@ import json
 import tempfile
 from pathlib import Path
 
+import pytest
+
+pytest.importorskip("pydantic")
+
 from kicad_tools.mcp.server import create_server
 from kicad_tools.mcp.tools.export import (
     ASSEMBLY_MANUFACTURERS,

--- a/tests/test_mcp_bom.py
+++ b/tests/test_mcp_bom.py
@@ -6,6 +6,10 @@ import json
 import tempfile
 from pathlib import Path
 
+import pytest
+
+pytest.importorskip("pydantic")
+
 from kicad_tools.mcp.server import create_server
 from kicad_tools.mcp.tools.export import (
     SUPPORTED_BOM_FORMATS,

--- a/tests/test_mcp_context.py
+++ b/tests/test_mcp_context.py
@@ -8,6 +8,8 @@ from __future__ import annotations
 
 import pytest
 
+pytest.importorskip("pydantic")
+
 from kicad_tools.mcp.context import (
     AgentPreferences,
     Decision,

--- a/tests/test_mcp_e2e.py
+++ b/tests/test_mcp_e2e.py
@@ -25,6 +25,8 @@ from pathlib import Path
 
 import pytest
 
+pytest.importorskip("pydantic")
+
 from kicad_tools.mcp.server import MCPServer, create_server
 from kicad_tools.mcp.tools.analysis import analyze_board, get_drc_violations
 from kicad_tools.mcp.tools.session import (

--- a/tests/test_mcp_export.py
+++ b/tests/test_mcp_export.py
@@ -9,6 +9,8 @@ from unittest.mock import MagicMock, patch
 
 import pytest
 
+pytest.importorskip("pydantic")
+
 from kicad_tools.mcp.server import create_server
 from kicad_tools.mcp.tools.export import (
     SUPPORTED_MANUFACTURERS,

--- a/tests/test_mcp_placement_analyze.py
+++ b/tests/test_mcp_placement_analyze.py
@@ -6,6 +6,8 @@ from pathlib import Path
 
 import pytest
 
+pytest.importorskip("pydantic")
+
 from kicad_tools.exceptions import FileNotFoundError as KiCadFileNotFoundError
 from kicad_tools.exceptions import ParseError
 from kicad_tools.mcp.tools.placement import placement_analyze

--- a/tests/test_mcp_routing.py
+++ b/tests/test_mcp_routing.py
@@ -4,6 +4,8 @@ from pathlib import Path
 
 import pytest
 
+pytest.importorskip("pydantic")
+
 from kicad_tools.exceptions import FileNotFoundError as KiCadFileNotFoundError
 from kicad_tools.exceptions import ParseError
 from kicad_tools.mcp.tools.routing import get_unrouted_nets, route_net

--- a/tests/test_mcp_session.py
+++ b/tests/test_mcp_session.py
@@ -8,6 +8,8 @@ from pathlib import Path
 
 import pytest
 
+pytest.importorskip("pydantic")
+
 from kicad_tools.mcp.tools.session import (
     apply_move,
     commit_session,

--- a/tests/test_mcp_session_intent.py
+++ b/tests/test_mcp_session_intent.py
@@ -8,6 +8,8 @@ from pathlib import Path
 
 import pytest
 
+pytest.importorskip("pydantic")
+
 from kicad_tools.mcp.tools.session import (
     apply_move,
     clear_intent,

--- a/tests/test_mcp_session_manager.py
+++ b/tests/test_mcp_session_manager.py
@@ -8,6 +8,8 @@ from pathlib import Path
 
 import pytest
 
+pytest.importorskip("pydantic")
+
 from kicad_tools.exceptions import FileNotFoundError as KiCadFileNotFoundError
 from kicad_tools.mcp.errors import (
     SessionNotFoundError,

